examples/js_callback/trunk_post_build.rs

```diff
@@ -0,0 +1,43 @@
+use std::fs;
+use std::io::{Read, Write};
+
+fn read_env(env: &'static str) -> String {
+    std::env::var(env).unwrap_or_else(|e| panic!("can't read {} env var: {}", env, e))
+}
+
+fn main() {
+    let stage_dir = read_env("TRUNK_STAGING_DIR");
+    let mut res = fs::read_dir(format!("{stage_dir}/snippets")).expect("no snippets dir in stage");
+    let dir = res
+        .next()
+        .expect("there must be one snippets dir present")
+        .expect("can't read snippets dir");
+    let dir_name = dir.file_name().to_string_lossy().to_string();
+    let mut index_html =
+        fs::File::open(format!("{stage_dir}/index.html")).expect("can't open index.html");
+    let mut html = String::new();
+    index_html
+        .read_to_string(&mut html)
+        .expect("can't read index.html");
+
+    let mut split = html
+        .split("</head>")
+        .map(|it| it.to_string())
+        .collect::<Vec<String>>();
+
+    let public_url = read_env("TRUNK_PUBLIC_URL");
+    let public_url = public_url.strip_suffix("/").unwrap_or(&public_url);
+    let wasm_bindgen_snippets_path = format!("{public_url}/snippets/{dir_name}");
+
+    split.insert(1,  format!("<script>window.wasmBindgenSnippetsPath = '{wasm_bindgen_snippets_path}';</script></head>"));
```

Comment:
Instead of modifying `index.html`, you could have the following block in `bindings.rs`

```rust
#[wasm_bindgen(raw_module = "./snippets_path_meta.js")]
extern "C" {
    type Global; // fake class. A bit wonky, but also how js_sys gets its ref to the global object
    #[wasm_bindgen(getter, static_method_of = Global, js_class = bindings)]
    pub fn snippets_path() -> String;
}
```

and then write `snippets_path_meta.js` into the `{stage_dir}` here. wasm_bindgen will insert the correct the correct import in the `index-*.js` glue code, and import the thing. `snippets_path_meta.js` should contain something like

```js
export const bindings = {
    snippets_path: "/snippets/js_callback-12fde9d6e52a7cb5"
};
```

Replaces

```
    #[wasm_bindgen(method, getter, js_name = "wasmBindgenSnippetsPath")]
    pub fn wasm_bindgen_snippets_path(this: &Window) -> String;
```

Reply:
That means one more request. We're trying to reduce the number of requests. It's going to be a less than a kilobyte transferred but still a request made none the less

Reply:
Fair enough. I had the insane thought of using some kind of `UUID` for the `raw_module = ` path and replacing it in the index-*.js glue with an data URI, like `data:text/javascript;charset=utf-8;base64,ZXhw...` (yes this works, importing from a data URI is fine) but that might be a bit ... much :)

If I understand everything correctly, the glue code is loaded with `<script type="module" ...>` and hence by default `defer`, so guaranteed to observe the `wasmBindgenSnippetsPath` the `trunk_post_build` insert.